module6/TaskListApp/TaskListApp/Views/TaskListView.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  TaskListView.swift
+//  TaskListApp
+//
+//  Created by Peter Rice on 12/22/23.
+//
+
+import SwiftUI
+
+struct TaskListView: View {
+  @StateObject var tasksStore = TasksStore()
+  @State var showingAddTaskView: Bool = false
+  @State var selectedTab = 0
+  @State var searchTaskName: String = ""
+  
+  var uncompletedTasks: [Task] {
+    if !searchTaskName.isEmpty {
+      return tasksStore.searchUncompletedTasks(searchTerm: searchTaskName)
+    }
+    return tasksStore.uncompletedTasks
+  }
+  
+  var completedTasks: [Task] {
+    if !searchTaskName.isEmpty {
+      return tasksStore.searchCompletedTasks(searchTerm: searchTaskName)
+    }
+    return tasksStore.completedTasks
+    
+  }
+  
+  var body: some View {
+    NavigationStack {
+      TabView(selection: $selectedTab) {
+        InlineTaskListView(tasks: uncompletedTasks, tasksStore: tasksStore)
+          .tabItem {
+            Image(systemName: "list.bullet.circle")
+              .resizable()
+            Text("Tasks")
+          }
+          .tag(0)
+        InlineTaskListView(tasks: completedTasks, tasksStore: tasksStore)
+          .tabItem {
+            Image(systemName: "checkmark.circle")
+              .resizable()
+            Text("Completed")
+          }
+          .tag(1)
+        InlineTaskCategoryView(vm: InlineTaskCategoryViewModel(tasksStore: tasksStore))
+          .tabItem {
+            Image(systemName: "tag.circle")
+              .resizable()
+            Text("Categories")
+          }
+          .tag(2)
+      }
+      .navigationTitle(Text("My Tasks"))
+      .navigationBarItems(
+        trailing: Button(action: {
+          showingAddTaskView = true
+        }, label: {
+          Image(systemName: "plus.circle.fill")
+            .bold()
+        }))
+      .navigationDestination(for: Task.self) { task in
+        TaskDetailsView(tasksStore: tasksStore, task: task)
+      }
+      .fullScreenCover(isPresented: $showingAddTaskView, content: {
+        AddTaskView(tasksStore: tasksStore)
+      })
+      .searchable(text: $searchTaskName, prompt: "Task Name")
+      
+    }
+  }
+}
+
+struct InlineTaskListView: View {
+  let tasks: [Task]
+  let tasksStore: TasksStore
+  
+  var body: some View {
+    List(tasks) { task in
+      NavigationLink(value: task) {
+        TaskRowView(task: task, tasksStore: tasksStore)
+      }
+    }
+    .listStyle(.plain)
+  }
```

Comment:
consider also having an animation here like `.animation(.default, value: tasks)` so that the deletion of a task animates as well. right now it's a little jarring.

Reply:
good suggestion. this is much better.